Initial content about containerized proxy (#1542)
uyuni-project/uyuni-docs: modules/installation-and-upgrade/pages/proxy-container-installation.adoc

[[installation-proxy-containers]]
= Prepare container host for {productname} Proxy containers

[NOTE]
====
Only {sles} 15 SP3 and newer are supported to be used as container host for {productname} Proxy containers.
====

[[installation-proxy-containers-requirements]]
== SLES Container Host Requirements

[cols="1,1,1", options="header"]
.Proxy Container Host Hardware Requirements
|===

| Hardware
| Details
| Recommendation

| CPU
|
| Minimum 2 dedicated 64-bit CPU cores

| RAM
| Test Server
| Minimum 2{nbsp}GB

|
| Production Server
| Minimum 8{nbsp}GB

| Disk Space
|
| Minimum 100{nbsp}GB

|===

[[installation-proxy-containers-services]]
== Install Container Services on SLES system

[NOTE]
====

> Container host used as a base for {productname} Proxy containers needs to be first registered as a salt client to the {productname} Server.

Reviewer: ```suggestion
Container host used as a base for {productname} Proxy containers needs to be first registered as a Salt client to the {productname} Server.
```

> For more information how to register salt client to the {productname} Server see xref:client-configuration:registration-overview.adoc[].

Reviewer: ```suggestion
For more information about registering Salt client to the {productname} Server, see xref:client-configuration:registration-overview.adoc[].
```

====

[NOTE]
====
[literal]``Containers Module`` is required to be available for container host.
====

{productname} Proxy containers are using [literal]``podman`` and [literal]``systemd`` to run and manage all proxy containers.

> First step is to install containers control files provided by package [literal]``uyuni-proxy-systemd-services``

Reviewer: ```suggestion
First step is to install container control files provided by package [literal]``uyuni-proxy-systemd-services``.
```

[[proc-installation-proxy-containers-services]]
.Procedure: Installation Of Container Services For {productname} Proxy

. Assign [literal]``Containers Module`` software channel to the container host in the {productname}.
+

> For information how to assign software channels to the system see xref:administration:channel-management.adoc[].

Reviewer: ```suggestion
For more information about assigning software channels to the system, see xref:administration:channel-management.adoc[].
```

. Log in as `root` on the container host.

. Manually install {productname} Proxy service package:
+

----
zypper install uyuni-proxy-systemd-services
----

[[installation-proxy-containers-customize-config]]
== Customize {productname} Proxy configuration

{productname} Proxy containers require some volumes to be mounted for long term storage.
Paths on container host for those volumes are by default:

> - [path]`/var/lib/uyuni/proxy-squid-cache`
> - [path]`/var/lib/uyuni/proxy-rhn-cache`
> - [path]`/var/lib/uyuni/proxy-tftpboot`

Reviewer: ```suggestion
- [path]``/var/lib/uyuni/proxy-squid-cache``
- [path]``/var/lib/uyuni/proxy-rhn-cache``
- [path]``/var/lib/uyuni/proxy-tftpboot``
```
I'm not sure whether duple backtics are needed but we usually do it that way.

> To modify location of those paths, please edit [path]`/etc/sysconfig/uyuni-proxy-systemd-services.config`.

Reviewer: ```suggestion
To modify location of those paths, edit [path]``/etc/sysconfig/uyuni-proxy-systemd-services.config``.
```

In this file it is possible to add custom arguments passed to podman container pod:

> - [literal]`EXTRA_POD_ARGS=''`

Reviewer: ```suggestion
- [literal]``EXTRA_POD_ARGS=''``
```

In this file it is possible to modify path where proxy configuration is expected, however modifying it is strongly discouraged:

> - [literal]`CONFIG_DIR`=[path]`/etc/uyuni/proxy`

Reviewer: ```suggestion
- [literal]``CONFIG_DIR`=[path]`/etc/uyuni/proxy``
```

== SLES Container Host Requirements

[cols="1,1,1", options="header"]
.Proxy Container Host Hardware Requirements
|===

| Hardware
| Details
| Recommendation

| CPU
|
| Minimum 2 dedicated 64-bit CPU cores

| RAM
| Test Server
| Minimum 2{nbsp}GB

|
| Production Server
| Minimum 8{nbsp}GB

| Disk Space
|
| Minimum 100{nbsp}GB

|===

[[installation-proxy-containers-services]]
== Install Container Services on SLES system

[NOTE]
====
Container host used as a base for {productname} Proxy containers needs to be first registered as a salt client to the {productname} Server.

For more information how to register salt client to the {productname} Server see xref:client-configuration:registration-overview.adoc[].
====

{productname} Proxy containers are using [literal]``podman`` and [literal]``systemd`` to run and manage all proxy containers.

First step is to install containers control files provided by package [literal]``uyuni-proxy-systemd-services``

[[proc-installation-proxy-containers-services]]
.Procedure: Installation Of Container Services For {productname} Proxy

. Log in as `root` on the container host.

. Install {productname} Proxy service package:
+

----

> zypper install uyuni-proxy-systemd-services

Author: Added requirement to assign containers module channel.

----

[[installation-proxy-containers-customize-config]]
== Customize {productname} Proxy configuration

{productname} Proxy containers require some volumes to be mounted for long term storage.
Paths on container host for those volumes are by default:

- [path]`/var/lib/uyuni/proxy-squid-cache`
- [path]`/var/lib/uyuni/proxy-rhn-cache`
- [path]`/var/lib/uyuni/proxy-tftpboot`

To modify location of those paths, please edit [path]`/etc/sysconfig/uyuni-proxy-systemd-services.config`.

In this file it is possible to add custom arguments passed to podman container pod:

- [literal]`EXTRA_POD_ARGS=''`

In this file it is possible to modify path where proxy configuration is expected, however modifying it is strongly discouraged:

- [literal]`CONFIG_DIR`=[path]`/etc/uyuni/proxy`

[[installation-proxy-containers-control-plane-ssh]]
== Set management SSH listening port on different then 22

> One of the containers providing {productname} Proxy services is a SSH container. This container main task is to provide bridge between {productname} Server and SSH Salt clients over isolated networks. This container however has to listen on standard SSH port 22/TCP. It is therefor required that if SSH service is used on container host its listening port must be different then 22/TCP.

Author: Removed
